code/game/objects/structures/crates_lockers/closets/job_closets.dm

```diff
@@ -56,6 +56,7 @@
 
 /obj/structure/closet/jcloset/New()
 	..()
+	new /obj/item/pornmag
```

Comment:
you forgot (src) at the end which is used to give a location to the pornmag(aka the closet itself)